syntax = "proto2";

package indexer_models;

message NFTMarketplaceActivityRow {
    optional int64 transaction_version = 1;
    optional int64 event_index = 2;
    optional string event_type = 3;
    optional string standard_event_type = 4;
    optional string creator_address = 5;
    optional string collection = 6;
    optional string token_name = 7;
    optional string token_data_id = 8;
    optional string collection_id = 9;
    optional int64 price = 10;
    optional int64 amount = 11;
    optional string buyer = 12;
    optional string seller = 13;
    optional string json_data = 14;
    optional string marketplace = 15;
    optional string contract_address = 16;
    optional string entry_function_id_str = 17;
    optional int64 transaction_timestamp = 18;
    optional int64 inserted_at = 19;
    optional string _CHANGE_TYPE = 20;
}